lib/rubygems/sigstore/rekor/api.rb

```diff
@@ -32,23 +32,11 @@ def create(cert_chain, data)
   end
 
   def where(data_digest:)
-    retrieve_response = connection.post("/api/v1/index/retrieve",
-      {
-        hash: "sha256:#{data_digest}",
-      }
-    )
+    log_entry_uuids = find_log_entry_uuids_by_digest(data_digest)
 
-    unless retrieve_response.status == 200
-      raise "Unexpected response from POST /api/v1/index/retrieve:\n #{retrieve_response}"
-    end
-
-    retrieve_response = connection.post("api/v1/log/entries/retrieve", entryUUIDs: retrieve_response.body)
+    return [] if log_entry_uuids.empty?
 
-    unless retrieve_response.status == 200
-      raise "Unexpected response from POST api/v1/log/entries/retrieve:\n #{entry_response}"
-    end
-
-    retrieve_response.body.reduce(:merge).map do |uuid, entry|
+    find_log_entries_by_uuid(log_entry_uuids).reduce({}, :merge).map do |uuid, entry|
```

Comment:
Using `{}` as the initial value in [`Enumerable#reduce`](https://ruby-doc.org/core-2.7.0/Enumerable.html#method-i-reduce) ensures that empty responses from `find_log_entries_by_uuid` will not produce a `nil` value heading into `map`.